pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -621,35 +625,82 @@ protected void internalCreatePartitionedTopic(AsyncResponse asyncResponse, int n
+            if (cluster.equals(pulsar().getConfiguration().getClusterName())) {
+                continue;
+            }
+            ClusterResources clusterResources = pulsar().getPulsarResources().getClusterResources();
+            CompletableFuture<Void> createRemoteTopicFuture = new CompletableFuture<>();
+            tasksForAllClusters.put(cluster, createRemoteTopicFuture);
+            clusterResources.getClusterAsync(cluster).whenComplete((clusterData, ex1) -> {
+                if (ex1 != null) {
+                    // Unexpected error, such as NPE. Catch all error to avoid the "createRemoteTopicFuture" stuck.
+                    log.error("[{}] An un-expected error occurs when trying to create partitioned topic {} in cluster"
+                                    + " {}.", clientAppId(), topicName, cluster, ex1);
+                    createRemoteTopicFuture.completeExceptionally(new RestException(ex1));
+                    return;
+                }
+                // Get cluster data success.
+                TopicsImpl topics =
+                        (TopicsImpl) pulsar().getBrokerService().getClusterPulsarAdmin(cluster, clusterData).topics();
+                topics.createPartitionedTopicAsync(shortTopicName, numPartitions, true, null)
+                        .whenComplete((ignore, ex2) -> {
+                    if (ex2 == null) {
+                        // Create success.
+                        log.info("[{}] Successfully created partitioned topic {} in cluster {}",
+                                clientAppId(), topicName, cluster);
+                        createRemoteTopicFuture.complete(null);
+                        return;
+                    }
+                    // Create topic on the remote cluster error.
+                    Throwable unwrapEx2 = FutureUtil.unwrapCompletionException(ex2);
+                    // The topic has been created before, check the partitions count is expected.
+                    if (unwrapEx2 instanceof PulsarAdminException.ConflictException) {
+                        topics.getPartitionedTopicMetadataAsync(shortTopicName).whenComplete((topicMeta, ex3) -> {
+                            if (ex3 != null) {
+                                // Unexpected error, such as NPE. Catch all error to avoid the
+                                // "createRemoteTopicFuture" stuck.
+                                log.error("[{}] Failed to check remote-cluster's topic metadata when creating"
+                                                + " partitioned topic {} in cluster {}.",
+                                        clientAppId(), topicName, cluster, ex3);
+                                createRemoteTopicFuture.completeExceptionally(new RestException(ex3));
+                            }
+                            // Call get partitioned metadata of remote cluster success.
+                            if (topicMeta.partitions == numPartitions) {
```

Comment:
Maybe we can improve the method `Topics#createPartitionedTopicAsync(String topic, int numPartitions, Map<String, String> properties)` to make it provide an error code for the partition count conflict problem, if so we don't need to check the partition count.

Reply:
Good suggestion, since only the latest releases(or latest patch releases) will contain the new error code we added, these compatibility codes are still needed. 

We can separate a PR to add the new error code and remove these compatibility codes after a long time.

Reply:
Got it, thanks.